### PR TITLE
fixes for the template plugin page

### LINKED ISSUE #492: Problem with page plugins/template.md
More example would be useful for custom template.

Missing also example of making snippet:
https://www.tinymce.com/docs/plugins/template/#makingsnippets

And here for custom variable there is not explained how to render them using {$varName}: https://www.tinymce.com/docs/plugins/template/#example

I just found on bottom under making snippet, but without that I was trying how to render variable "username" and "staffid" of the examples.

> *affected page: https://www.tinymce.com/docs/plugins/template/*

### PR DESCRIPTION
**Pull request by tylerkelly13**
Fixes #492

### CODE REVIEW

File: plugins/template.md [hunk 1]
```diff
@@ -353,8 +351,27 @@ Each template needs to be inside of a div with the `mceTmpl` class, like this ex
 
 ### Making Snippets
 
-Snippets are `html` code chunks that can be inserted. Replace variables will only be executed upon insert, without being wrapped in a template `div` element. So if you define `somevar1` in `template_replace_values` array it will be replaced on preview and insert.
+Snippets are `html` code chunks that can be inserted. Replace variables will only be executed upon insert, without being wrapped in a template `div` element. So if you define a variable in with [`template_replace_values`](#template_replace_values) it will be replaced on insert.
```
Comment (metricjs) [inline plugins/template.md]: "in with"?

File: plugins/template.md [hunk 2]
```diff
@@ -329,9 +329,7 @@ This is the contents your backend page should return if you specify a URL in the
 
 ### Making Templates
```
Comment (metricjs) [inline plugins/template.md]: Should these headings become something like "Making Template Files" and "Making Inline Templates"? Idk what language we use elsewhere but these mean basically nothing to me

Comment (metricjs) [reply plugins/template.md]: And reference the content/url thing or whatever it is?

File: plugins/template.md (hunk 1 above)
Comment (tylerkelly13) [reply plugins/template.md]: fixed

File: plugins/template.md (hunk 2 above)
Comment (tylerkelly13) [reply plugins/template.md]: fixed

**Pull request merged**